compiler/src/dotty/tools/dotc/core/ContextOps.scala

```diff
@@ -5,11 +5,45 @@ import Contexts._, Symbols._, Types._, Flags._
 import Denotations._, SymDenotations._
 import Names.Name, StdNames.nme
 import ast.untpd
+import scala.util.control.NonFatal
 
 /** Extension methods for contexts where we want to keep the ctx.<methodName> syntax */
 object ContextOps:
-
+  case class Implosion(ex: Throwable) extends Exception(ex)
```

Comment:
I think `Implosion` should be an [`Error`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Error.html), although that won't be enough to pass through the compiler cleanly since there's many places where we catch all `Throwable` or use the `NonFatal` extractor

Reply:
I think `Error` probably makes sense. Per previous [comment](https://github.com/lampepfl/dotty/pull/16593#issuecomment-1366459339) I think the use of `Throwable` & the `NonFatal` extractor should be replaced with something specific to the compiler which won't catch `ControlFlow` exceptions, so this wouldn't be a problem

Reply:
Changed to `ControlThrowable` instead of `Error` to avoid being caught by existing `NonFatal` extractors